dialer_test.go

```diff
@@ -271,6 +285,65 @@ func TestIAMAuthNErrors(t *testing.T) {
 	}
 }
 
+func TestUniverseDomain(t *testing.T) {
+	tcs := []struct {
+		desc string
+		opts Option
+	}{
+		{
+			desc: "When universe domain matches GDU",
+			opts: WithOptions(WithUniverseDomain("googleapis.com"), WithCredentialsJSON(fakeServiceAccount)),
+		},
+		{
+			desc: "When universe domain from admin api matches GDU",
+			opts: WithOptions(WithAdminAPIEndpoint("https:sqladmin.googleapis.com"), WithCredentialsJSON(fakeServiceAccount)),
+		},
+		{
+			desc: "When TPC universe matches TPC credential domain",
+			opts: WithOptions(WithUniverseDomain("test-universe.test"), WithCredentialsJSON(fakeTPCServiceAccount)),
```

Comment:
Maybe fakeTPCServiceAccount should be a function that takes a domain? That makes the relationship between inputs here more clear.

Reply:
I might do this as a follow up PR, ideally i should just have `fakeServiceAccount` take in a string for universe domain and then i wouldn't even need `fakeTPCServiceAccount`